qiskit/backends/ibmq/ibmqbackend.py

```diff
@@ -214,10 +214,15 @@ def jobs(self, limit=50, skip=0, status=None, db_filter=None):
         job_list = []
         for job_info in job_info_list:
             is_device = not bool(self._configuration.get('simulator'))
+            if 'header' in job_info:
+                backend_name = job_info['header'].get('backend_name')
+            elif 'backend' in job_info:  # old style job_info
+                backend_name = job_info['backend'].get('name')
             job = IBMQJob(self._api, is_device,
                           job_id=job_info.get('id'),
-                          backend_name=job_info.get('backend').get('name'),
-                          creation_date=job_info.get('creationDate'))
+                          backend_name=backend_name,
+                          creation_date=job_info.get('creationDate'),
+                          backend_allows_qobj=self.configuration().get('allows_q_object'))
```

Comment:
This sounds ok to me. I'm not sure about calling it Deprecated however since currently it is possible to retrieve past jobs in the old format even after the backends have been updated. We may not be able to ever remove creating "deprecated" jobs. The fontend calls them qasms jobs since they are closer to that format but this may be confusing too.